lib/mongo/collection.rb

```diff
@@ -372,13 +372,13 @@ def capped?
     # @since 2.0.0
     def create(opts = {})
       # Passing read options to create command causes it to break.
-      # Filter the read options out.
+      # Filter the read options out. The same is true with session
+      # since it gets passed to the operation and fails BSON serialization.
```

Comment:
Unless I'm misunderstanding the situation, the correct comment here would be that session is handled further below on line 391. It is ultimately passed to the server just using a different code path.